src/NuGet.Core/NuGet.PackageManagement/BuildIntegration/BuildIntegratedRestoreUtility.cs

```diff
@@ -174,8 +175,21 @@ await project
                     if (dependency?.SuppressParent == LibraryIncludeFlagUtils.DefaultSuppressParent &&
                         dependency?.IncludeType == LibraryIncludeFlags.All)
                     {
+                        var includeType = LibraryIncludeFlags.All & ~LibraryIncludeFlags.Compile;
                         dependency.SuppressParent = LibraryIncludeFlags.All;
-                        dependency.IncludeType = LibraryIncludeFlags.All & ~LibraryIncludeFlags.Compile;
+                        dependency.IncludeType = includeType;
+
+                        // update lock file target libraries
+                        foreach (var target in lockFile.Targets
```

Comment:
I don't think this logic belongs here. 

The LockFileBuilder seems like the best place to make these types of decisions. Decentralizing it makes it difficult to follow. 

Furthermore I am not sure in what way this is an improvement over the previous fix.
How would this flow to dev? Would you create a new PR etc.

Reply:
LockFileBuilder isn't the place to put it in since it will then impact the restore codepath as well which we don't want. Earlier I put it there itself as part of previous fix which is why this is the improvement of that fix. 

I'll cherry-pick this in dev.

Reply:
Can the logic then be contained in the same class but just invoked from here? 
Just think that every modification to the assets file should be contained within 1 class. 
Feel free to make a judgment call. 

Don't you need to remove part of the fix made in the first PR?

Reply:
I feel this is a special case of updating lock file for `developmentDependency` and all that change should be at one place. We're already updating `PackageSpec` here (which is also part of lock file) so it makes sense to update libraries. It will be easy for anyone to understand that at one place.

While cherry-picking this, I'll make sure to remove part of existing fix (since the tests changes are still relevant).

Reply:
That's fine. It was a suggestion/my personal preference, but I'm not gonna argue over it too much, as we don't have a well defined coding style.